pephubclient/modules/sample.py

```diff
@@ -0,0 +1,206 @@
+from pephubclient.exceptions import ResponseError
+
+_LOGGER = logging.getLogger("pephubclient")
+
+
+class PEPHubSample(RequestManager):
+    """
+    Class for managing samples in PEPhub and provides methods for
+        getting, creating, updating and removing samples.
+    This class is not related to peppy.Sample class.
+    """
+
+    def __init__(self, jwt_data: str = None):
+        """
+        :param jwt_data: jwt token for authorization
+        """
+
+        self.__jwt_data = jwt_data
+
+    def get(
+        self,
+        namespace: str,
+        name: str,
+        tag: str,
+        sample_name: str = None,
+    ) -> dict:
+        """
+        Get sample from project in PEPhub.
+
+        :param namespace: namespace of project
+        :param name: name of project
+        :param tag: tag of project
+        :param sample_name: sample name
+        :return: Sample object
+        """
+        url = self._build_sample_request_url(
+            namespace=namespace, name=name, sample_name=sample_name
+        )
+
+        url = url + self.parse_query_param(pep_variables={"tag": tag})
+
+        response = self.send_request(
+            method="GET", url=url, headers=self.parse_header(self.__jwt_data)
+        )
+        if response.status_code == ResponseStatusCodes.OK:
+            return self.decode_response(response, output_json=True)
+        if response.status_code == ResponseStatusCodes.NOT_EXIST:
+            raise ResponseError("Sample does not exist.")
+        elif response.status_code == ResponseStatusCodes.INTERNAL_ERROR:
+            raise ResponseError("Internal server error. Unexpected return value.")
+        else:
+            raise ResponseError(
+                f"Unexpected return value. Error: {response.status_code}"
+            )
+
+    def create(
+        self,
+        namespace: str,
+        name: str,
+        tag: str,
+        sample_name: str,
+        sample_dict: dict,
+        overwrite: bool = False,
+    ) -> None:
+        """
+        Create sample in project in PEPhub.
+
+        :param namespace: namespace of project
+        :param name: name of project
+        :param tag: tag of project
+        :param sample_dict: sample dict
+        :param sample_name: sample name
+        :param overwrite: overwrite sample if it exists
+        :return: None
+        """
+        url = self._build_sample_request_url(
+            namespace=namespace,
+            name=name,
+            sample_name=sample_name,
+        )
+
+        url = url + self.parse_query_param(
+            pep_variables={"tag": tag, "overwrite": overwrite}
+        )
+
+        # add sample name to sample_dict if it is not there
+        if sample_name not in sample_dict.values():
+            sample_dict["sample_name"] = sample_name
+
+        response = self.send_request(
+            method="POST",
+            url=url,
+            headers=self.parse_header(self.__jwt_data),
```

Comment:
I wonder if we could just set `headers` to `None` and then it calls `self.parse_header` on the inside?

Reply:
Nvm, we are in `self` so I think this is fine